tests/unit/amo/test_imageUtils.js

```diff
@@ -1,19 +1,101 @@
 import { getAddonIconUrl, getPreviewImage } from 'amo/imageUtils';
-import {
-  createInternalAddonWithLang,
-  fakeAddon,
-  fakePreview,
-} from 'tests/unit/helpers';
+import { createInternalAddonWithLang, fakePreview } from 'tests/unit/helpers';
 import fallbackIcon from 'amo/img/icons/default.svg';
 
-describe(__filename, () => {
-  const allowedIcon = 'https://addons.mozilla.org/webdev-64.png';
+const iconsUrlsExample = {
+  small: 'https://addons.mozilla.org/get-icon-exampele-sml-32.png',
+  regular: 'https://addons.mozilla.org/get-icon-example-reg-64.png',
+  large: 'https://addons.mozilla.org/get-icon-example-lrg-128.png',
+};
 
+describe(__filename, () => {
   describe('getAddonIconUrl', () => {
-    it('return icon url as in fake addon', () => {
-      expect(getAddonIconUrl({ ...fakeAddon, icon_url: allowedIcon })).toEqual(
-        allowedIcon,
+    it('throw error if requested icon size is invalid', () => {
+      const getIconWithInvalidSize = () => getAddonIconUrl({}, 25);
+
+      expect(getIconWithInvalidSize).toThrowErrorMatchingInlineSnapshot(
+        `"size must be one of: 32,64,128"`,
+      );
+    });
```

Comment:
the expected result would be it would fallback to the default icon size, 64.

Reply:
Changed